.github/workflows/validate.yml

```diff
@@ -0,0 +1,41 @@
+name: Validate
+on:
+  pull_request:
+jobs:
+  lint:
+    name: Run eslint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 20
+      - name: Install dependencies
+        run: yarn
+      - name: run eslint
+        run: yarn lint
+  tsc:
+    name: Run tsc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 20
+      - name: Install dependencies
+        run: yarn
+      - name: run tsc
+        run: yarn tsc
+  build:
+    name: Run Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 20
+      - name: Install dependencies
```

Comment:
change: i don't see much of use of the multiple `Install dependencies`? can you please elaborate why this is needed

Reply:
dependencies are required for each job to run successfully.